src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -170,21 +168,32 @@ CATCH_RETURN()
         UINT32 uiLengthRead = 0;
 
         // Start from the beginning.
-        const UINT32 glyphStart = 0;
+        UINT32 pos = 0;
 
         _glyphIndices.resize(textLength);
 
-        const HRESULT hr = _fontRenderData->Analyzer()->GetTextComplexity(
-            _text.c_str(),
-            textLength,
-            _fontInUse,
-            &isTextSimple,
-            &uiLengthRead,
-            &_glyphIndices.at(glyphStart));
-
-        RETURN_IF_FAILED(hr);
+        while (pos < textLength)
+        {
+            const HRESULT hr = _fontRenderData->Analyzer()->GetTextComplexity(
+                _text.c_str() + pos,
+                textLength,
+                _fontInUse,
+                &isTextSimple,
+                &uiLengthRead,
+                &_glyphIndices.at(pos));
+
+            RETURN_IF_FAILED(hr);
+            _SetCurrentRun(pos);
+            _SplitCurrentRun(pos);
+            pos += uiLengthRead;

```

Comment:
This is going to sound super weird, but are we sure that `uiLengthRead` will always return >0 from `GetTextComplexity()`? 

I've been burned/surprised inside while loops that turn out infinite for weird things like that before.